src/main/java/edu/indiana/dlib/amppd/model/factory/ObjectFactory.java

```diff
@@ -0,0 +1,56 @@
+package edu.indiana.dlib.amppd.model.factory;
+
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.lang.reflect.*;
+
+import edu.indiana.dlib.amppd.model.Bundle;
+import edu.indiana.dlib.amppd.model.Collection;
+import edu.indiana.dlib.amppd.model.CollectionSupplement;
+import edu.indiana.dlib.amppd.model.Dataentity;
+import edu.indiana.dlib.amppd.model.Item;
+import edu.indiana.dlib.amppd.model.ItemSupplement;
+import edu.indiana.dlib.amppd.model.Job;
+import edu.indiana.dlib.amppd.model.JobMgmMode;
+import edu.indiana.dlib.amppd.model.Mgm;
+import edu.indiana.dlib.amppd.model.MgmMode;
+import edu.indiana.dlib.amppd.model.MgmModeInput;
+import edu.indiana.dlib.amppd.model.MgmModeOutput;
+import edu.indiana.dlib.amppd.model.Primaryfile;
+import edu.indiana.dlib.amppd.model.PrimaryfileSupplement;
+import edu.indiana.dlib.amppd.model.RouteLink;
+import edu.indiana.dlib.amppd.model.Unit;
+import edu.indiana.dlib.amppd.model.Workflow;
+
+
+public class ObjectFactory implements BaseObjectFactory{
+
+	@Override
+	public <O extends Dataentity> O createDataentityObject(HashMap<?,?> args, String classname) throws ClassNotFoundException
+	{
+		O res = null;
+		Class<?> entityObj = Class.forName("edu.indiana.dlib.amppd.model."+classname);
+		try 
+		{
+			res = (O)entityObj.getConstructor().newInstance();			
+			for(Entry<?, ?> entry : args.entrySet())
+			{ 
+				Field field = entityObj.getDeclaredField(entry.getKey().toString());
+				if(entry.getKey().toString().equals(field.getName()))
+				{
+					field.setAccessible(true);
+					field.set(res, entry.getValue());
+				}
+			}
+		}
+		catch(Exception e)
+		{
+			System.out.println("Exception occurred in getting the new instance"+e);
+		}
+		

```

Comment:
- are you catching all kinds of exceptions or just ClassNotFound exceptions? If latter, you better specify the exception. In any case, you probably want to rethrow a RuntimeException so it will propagate up instead of silently return without doing anything.
- could you use log.error instead of System.out.print? using log (for all kinds of print out) will enable all messages written to the log file, which will be harvested by the LogManager later.